types/next-server/test/next-server-dynamic-tests.tsx

```diff
@@ -18,29 +13,34 @@ const LoadingComponent: React.StatelessComponent<LoadingComponentProps> = ({
 
 // 1. Basic Usage (Also does SSR)
 const DynamicComponent = dynamic(asyncComponent);
-const dynamicComponentJSX = <DynamicComponent foo="bar" />;
+const dynamicComponentJSX = <DynamicComponent foo />;
 
 // 1.1 Basic Usage (Loader function)
 const DynamicComponent2 = dynamic(() => asyncComponent);
-const dynamicComponent2JSX = <DynamicComponent2 foo="bar" />;
+const dynamicComponent2JSX = <DynamicComponent2 foo />;
 
 // 2. With Custom Loading Component
-const DynamicComponentWithCustomLoading = dynamic(asyncComponent, {
+const DynamicComponentWithCustomLoading = dynamic(() => asyncComponent, {
     loading: LoadingComponent
 });
-const dynamicComponentWithCustomLoadingJSX = <DynamicComponentWithCustomLoading foo="bar" />;
+const dynamicComponentWithCustomLoadingJSX = <DynamicComponentWithCustomLoading foo />;
 
 // 3. With No SSR
-const DynamicComponentWithNoSSR = dynamic(asyncComponent, {
+const DynamicComponentWithNoSSR = dynamic(() => asyncComponent, {
     ssr: false
 });
 
 // 4. With Multiple Modules At Once
-const HelloBundle = dynamic<MyComponentProps>({
+// TODO: Mapped components still doesn't infer their props.
```

Comment:
This is a difficult one. We can still pass `BundleComponentProps` into the root component, however the internally mapped components (in the `modules` property) doesn't map in the props from the dynamically-imported component.

So in the `render` property we can do:
```tsx
{
    render: (props, { Hello1, Hello2 }) => (
        <div>
            <h1>{props.foo}</h1>
            <Hello1 />
            <Hello2 />
        </div>
    )
}
```

but we can't use the props from `asyncComponent` since it's mapped as `React.ComponentType<{}>`

Reply:
That being said, the scope of this PR was only to fix the dynamic imports type error. I can try and solve this on a separate PR after this is merged.

Reply:
Yeah, I see the problem. But the modules map is dynamically generated.. not sure if that's possible at all, unless with some TS wizardry 😩